### PR TITLE
Change checkpointing train

### PR DESCRIPTION
**Pull request by NickleDave**

### CODE REVIEW

**Comment by NickleDave** [comment]
@yardencsGitHub I think this is working for me, can you check out the branch and test it with your data please?

**Comment by yardencsGitHub** [comment]
I checkout this branch and ran prep and train.
(I changed to val_step and ckpt_step and commented out the save_only_single_checkpoint option)


@NickleDave, I assume these parameters now pertain to steps and not epochs. 

Running 'train' with both checkpoint and validation steps set to 100:
First 100 steps seem fine.
Then, for the rest of the training phase there was an output line for each step and not a single line with progress %.. resulting in filling the screen. I set the number of epochs to be 1. At the end of the training steps 2 checkpoint files were saved but the training continued to run. I'm letting it continue but this is strange

**Comment by NickleDave** [comment]
Thank you for testing the changes.

Re: getting a new line for each step: sometimes the package that handles the progress bar does that (`tqdm`). It doesn't mean anything is going wrong with training. Sometimes you can "fix" it by resizing your terminal window, e.g. making it wider.

Can you clarify what you mean when you say that "at the end of training steps"? Do you mean after the last batch of the epoch?

Are you sure it's not just because you hit a validation step and then it went back to training after computing accuracy on the validation set?
That would explain why you now have two checkpoints *and* it is still training.

I'm going to run `vak` on `llb3` just to see if I run into any issues, should have thought of doing that last night to save you the effort.

**Comment by yardencsGitHub** [comment]
It was a validation point and it did go back to training. But this happened after >> the number of steps I set for validation.

On Apr 12, 2020, at 2:05 PM, David Nicholson <notifications@github.com> wrote:


Thank you for testing the changes.

Re: getting a new line for each step: sometimes the package that handles the progress bar does that (tqdm). It doesn't mean anything is going wrong with training. Sometimes you can "fix" it by resizing your terminal window, e.g. making it wider.

Can you clarify what you mean when you say that "at the end of training steps"? Do you mean after the last batch of the epoch?

Are you sure it's not just because you hit a validation step and then it went back to training after computing accuracy on the validation set?
That would explain why you now have two checkpoints and it is still training.

I'm going to run vak on llb3 just to see if I run into any issues, should have thought of doing that last night to save you the effort.

**Comment by NickleDave** [comment]
It should compute validation accuracy every `ckpt_step` number of steps, is that what happened?
I.e., you set `ckpt_step` to 200 so it also measured accuracy on step 400, 600, 800, etc.?

**Comment by yardencsGitHub** [comment]
I don’t think it does.
What is val_step then?

On Apr 12, 2020, at 3:09 PM, David Nicholson <notifications@github.com> wrote:


It should compute validation accuracy every ckpt_step number of steps, is that what happened?
I.e., you set ckpt_step to 200 so it also measured accuracy on step 400, 600, 800, etc.?

**Comment by NickleDave** [comment]
Sorry, I meant it should run validation every `val_step` steps, not `ckpt_step`.

What I'm trying to say is it's expected behavior to run the validation every `val_step` number of steps -- is that what it's doing?
It's still on the first validation step for me :/

**Comment by NickleDave** [comment]
I think this is working, going to go ahead and merge in.
